Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -3060,4 +3060,44 @@ class HTTPClientTests: XCTestCase {
         XCTAssertEqual(firstConnectionNumber, secondConnectionNumber, "Identical TLS configurations did not use the same connection")
         XCTAssertNotEqual(thirdConnectionNumber, firstConnectionNumber, "Different TLS configurations did not use different connections.")
     }
+
+    func testConnectionPoolSizeConfigValueIsRespected() {
+        let numberOfRequestsPerThread = 1000
+        let numberOfParallelWorkers = 32
+        let poolSize = 24
+
+        let httpBin = HTTPBin()
+        defer { XCTAssertNoThrow(try httpBin.shutdown()) }
+
+        let group = MultiThreadedEventLoopGroup(numberOfThreads: 4)
+        defer { XCTAssertNoThrow(try group.syncShutdownGracefully()) }
+
+        let configuration = HTTPClient.Configuration(
+            connectionPool: .init(concurrentHTTP1ConnectionsPerHostSoftLimit: poolSize)
+        )
+        let client = HTTPClient(eventLoopGroupProvider: .shared(group), configuration: configuration)
+        defer { XCTAssertNoThrow(try client.syncShutdown()) }
+
+        let g = DispatchGroup()
+        for workerID in 0..<numberOfParallelWorkers {
+            DispatchQueue(label: "\(#file):\(#line):worker-\(workerID)").async(group: g) {
+                func makeRequest() {
+                    let url = "http://127.0.0.1:\(httpBin.port)"
+                    XCTAssertNoThrow(try client.get(url: url).wait())
+                }
+                for _ in 0..<numberOfRequestsPerThread {
+                    makeRequest()
+                }
+            }
+        }
+        let timeout = DispatchTime.now() + .seconds(180)
```

Comment:
This timeout is _really_ long for CI. Can we lower it?

Reply:
I'm afraid we will need to change the number of requests and threads to something smaller, if we change the timeout to 60 seconds :)

```
Test Case 'HTTPClientTests.testConnectionPoolSizeConfigValueIsRespected' started at 2021-09-24 13:48:02.845
/code/Tests/AsyncHTTPClientTests/HTTPClientTests.swift:3101: error: HTTPClientTests.testConnectionPoolSizeConfigValueIsRespected : failed - Timed out
/code/Tests/AsyncHTTPClientTests/HTTPClientTests.swift:3089: error: HTTPClientTests.testConnectionPoolSizeConfigValueIsRespected : XCTAssertNoThrow failed: threw error "HTTPClientError.cancelled" - 
/code/Tests/AsyncHTTPClientTests/HTTPClientTests.swift:3089: error: HTTPClientTests.testConnectionPoolSizeConfigValueIsRespected : XCTAssertNoThrow failed: threw error "HTTPClientError.alreadyShutdown" - 
```

Reply:
Reduced the amount of requests.